### PR TITLE
Ibrahim/tab bar

### PR DESCRIPTION
**Pull request by Ibrahimhass**
Added support for tab-bar

### CODE REVIEW

File: lib/components/tab_bar/tab_bar.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    final colors = theme.colors;
+
+    return Align(
+      alignment: Alignment.topLeft,
+      child: Container(
+        decoration: ShapeDecoration(
+          color: colors.surfaceVariant,
+          shape: const StadiumBorder(),
+        ),
+        height: 40,
```
Comment (kalpeshp0310) [inline lib/components/tab_bar/tab_bar.dart:47]: In Figma, there are two sizes. "Regular" and "Large". this size is for the regular variant. we need to add support for both sizes.

File: lib/components/tab_bar/tab_bar.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  const TUITabBar({
+    super.key,
+    required this.titles,
+    required this.selectedIndex,
+    required this.onPressed,
+  });
+
+  Widget _buildItem(int index, String item) {
+    return TUIButton(
```
Comment (kalpeshp0310) [inline lib/components/tab_bar/tab_bar.dart:20]: For regular size, we have to use a 32px sized button, and for the large one, we have to use a 40px sized button.

**Comment by kalpeshp0310** [comment]
Please include the component in the Readme.

File: lib/components/button/button.dart [hunk 3]
```diff
@@ -127,7 +129,7 @@ class TUIButton extends StatelessWidget {
     }
 
     return Container(
-        margin: const EdgeInsets.all(5),
+        margin: customInsets ?? EdgeInsets.all(5),
```
[the remark this answers is not in the document]
Comment (Ibrahimhass) [reply lib/components/button/button.dart:132]: We are anyways going to refactor the button widget, next so let's deal with this at that time.

**Review by kalpeshp0310 — APPROVED** [review]

**Pull request merged**